model/src/payouts/quanto.rs

```diff
@@ -0,0 +1,609 @@
+            self.initial_price,
+            self.multiplier,
+        )
+        .context("Could not calculate liquidation thresholds")?;
+
+        Ok(0..=thresholds.long())
+    }
+
+    /// Price interval at which the party going short gets liquidated i.e. their payout amount
+    /// equals zero.
+    fn short_liquidation_interval(&self) -> Result<RangeInclusive<u64>> {
+        let initial_margin = self
+            .initial_margin_short()
+            .context("Could not calculate short's initial margin")?;
+        let initial_margin = initial_margin
+            .to_signed()
+            .context("Could not convert short's initial margin to bitcoin::SignedAmount")?;
+        let effective_initial_margin =
+            initial_margin + self.fee_offset.as_signed_amount(Position::Short);
+        let effective_initial_margin = effective_initial_margin
+            .to_unsigned()
+            .context("Could not convert short's effective initial margin to bitcoin::Amount")?;
+
+        let thresholds = LiquidationThresholds::new(
+            effective_initial_margin,
+            self.n_contracts,
+            self.initial_price,
+            self.multiplier,
+        )
+        .context("Could not calculate liquidation thresholds")?;
+
+        Ok(thresholds.short()..=maia_core::interval::MAX_PRICE_DEC)
+    }
+
+    /// Compute the initial BTC margin that the party going long has to put up.
+    fn initial_margin_long(&self) -> Result<Amount> {
+        calculate_initial_margin(
+            self.initial_price,
+            self.n_contracts,
+            self.leverage_long,
+            self.multiplier,
+        )
+        .context("Could not calculate long's initial margin")
+    }
+
+    /// Compute the initial BTC margin that the party going short has to put up.
+    fn initial_margin_short(&self) -> Result<Amount> {
+        calculate_initial_margin(
+            self.initial_price,
+            self.n_contracts,
+            self.leverage_short,
+            self.multiplier,
+        )
+        .context("Could not calculate short's initial margin")
+    }
+
+    /// Compute the profit and loss (PNL) at the given `closing_price`.
+    fn pnl_at_closing_price(&self, closing_price: u64) -> Result<Pnl> {
+        Pnl::new(
+            self.initial_price,
+            closing_price,
+            self.multiplier,
+            self.n_contracts,
+        )
+    }
+}
+
+/// Compute the initial BTC margin that a party has to put up, according to their `leverage`.
+fn calculate_initial_margin(
+    initial_price: u64,
+    n_contracts: u64,
+    leverage: Leverage,
+    multiplier: Decimal,
+) -> Result<Amount> {
+    let n_contracts = Decimal::from(n_contracts);
+    let leverage = Decimal::from(leverage.get());
+    let initial_price = Decimal::from(initial_price);
+
+    let margin = (n_contracts * initial_price * multiplier) / leverage;
+    let margin = margin.round_dp_with_strategy(8, RoundingStrategy::AwayFromZero);
```

Comment:
Are you sure we want `AwayFromZero`? Would `MidpointAwayFromZero` not be more _natural_?

Reply:
I definitely wasn't sure! I'm not too fussed, so `MidpointAwayFromZero` it is :D